app/scripts/directives/editPvc.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .directive("editPvc",
+             function($uibModal,
+                      $filter,
+                      $routeParams,
+                      APIService,
+                      DataService,
+                      ProjectsService,
+                      NotificationsService,
+                      Logger) {
+
+    return {
+      restrict: "E",
+      scope: {
+        // pvc object
+        pvc: "@",
+        // Alerts object for success and error alerts.
+        alerts: "=",
+        // Optional link label. Defaults to "Expand".
+        label: "@?",
+        // Optional redirect URL when the delete succeeds
+        redirectUrl: "@?",
+        // currentCapacity of the pvc
+        currentCapacity: "@"
+      },
+      template: '<a href="" ng-click="openEditModal()" role="button">{{label || "Expand PVC"}}</a>',
+      replace: true,
+      link: function(scope) {
+
+        scope.openEditModal = function() {
+          // opening the modal with settings scope as parent
+          var modalInstance = $uibModal.open({
+            templateUrl: 'views/modals/edit-resource.html',
+            controller: 'EditModalController',
+            scope: scope
+          });
+
+          scope.pvcVersion = APIService.getPreferredVersion('persistentvolumeclaims');
+
+          modalInstance.result.then(function(updatedSize) {
+            // upon clicking edit button, edit resource and send notification
+            ProjectsService
+            .get($routeParams.project)
+            .then(_.spread(function(project, context) {
+              scope.project = project;
+              scope.projectContext = context;
+              DataService
+                .get(scope.pvcVersion, $routeParams.pvc, context)
+                .then(function(pvc) {
+                  var updatedPvc = angular.copy(pvc);
+                  _.set(updatedPvc, ["spec", "resources", "requests", "storage"], updatedSize);
+                  var kind = pvc.kind;
+                  var resourceName = pvc.metadata.name;
+                  var typeDisplayName = scope.typeDisplayName || $filter('humanizeKind')(kind);
+                  var formattedResource = _.capitalize(typeDisplayName) + ' ' + "\'"  + (scope.displayName ? scope.displayName : resourceName) + "\'";
+                  var context = {namespace: pvc.metadata.namespace};
+
+                  DataService.update({
+                    resource: APIService.kindToResource(kind)
+                  }, resourceName, updatedPvc, context)
+                  .then(function() {
+                    NotificationsService.addNotification({
+                      type: "success",
+                      message: formattedResource + " expand request has been submitted."
+                    });
+                  })
+                  .catch(function(err) {
+                   // called if failure to edit
+                    NotificationsService.addNotification({
```

Comment:
The notification should have an ID so it can be cleared when the form is resubmitted. See

https://github.com/openshift/origin-web-console/blob/master/app/scripts/controllers/createPersistentVolumeClaim.js#L74